File: php/test/V2018DirectTest.php
<?php
declare(strict_types=1);

// V2018 direct test

require_once __DIR__ . '/../aareguru_sdk.php';
require_once __DIR__ . '/Runner.php';

use PHPUnit\Framework\TestCase;

class V2018DirectTest extends TestCase
{
    public function test_direct_load_v2018(): void
    {
        $setup = v2018_direct_setup(["id" => "direct01"]);
        $client = $setup["client"];


        [$result, $err] = $client->direct([
            "path" => "v2018/history",
            "method" => "GET",
            "params" => [],
        ]);
        $this->assertNull($err);
        $this->assertTrue($result["ok"]);
        $this->assertEquals(200, Helpers::to_int($result["status"]));
        $this->assertNotNull($result["data"]);

        if (!$setup["live"]) {
            if (is_array($result["data"]) && isset($result["data"]["id"])) {
                $this->assertEquals("direct01", $result["data"]["id"]);
            }
            $this->assertCount(1, $setup["calls"]);
        }
    }

}


function v2018_direct_setup($mockres)
{
    Runner::load_env_local();

    $calls = new \ArrayObject();

    $env = Runner::env_override([
        "AAREGURU_TEST_V_____ENTID" => [],
        "AAREGURU_TEST_LIVE" => "FALSE",
        "AAREGURU_APIKEY" => "NONE",
    ]);

    $live = $env["AAREGURU_TEST_LIVE"] === "TRUE";

    if ($live) {
        $merged_opts = [
            "apikey" => $env["AAREGURU_APIKEY"],
        ];
        $client = new AareguruSDK($merged_opts);
        return [
            "client" => $client,
            "calls" => $calls,
            "live" => true,
            "idmap" => [],
        ];
    }

    $mock_fetch = function ($url, $init) use ($calls, $mockres) {
        $calls[] = ["url" => $url, "init" => $init];
        return [
            [
                "status" => 200,
                "statusText" => "OK",
                "headers" => [],
                "json" => function () use ($mockres) {
                    if ($mockres !== null) {
                        return $mockres;
                    }
                    return ["id" => "direct01"];
                },
                "body" => "mock",
            ],
            null,
        ];
    };

    $client = new AareguruSDK([
        "base" => "http://localhost:8080",
        "system" => [
            "fetch" => $mock_fetch,
        ],
    ]);

    return [
        "client" => $client,
        "calls" => $calls,
        "live" => false,
        "idmap" => [],
    ];
}

File: rb/test/v2018_direct_test.rb
# V2018 direct test

require "minitest/autorun"
require "json"
require_relative "../Aareguru_sdk"
require_relative "runner"

class V2018DirectTest < Minitest::Test
  def test_direct_load_v2018
    setup = v2018_direct_setup({ "id" => "direct01" })
    client = setup[:client]


    result, err = client.direct({
      "path" => "v2018/history",
      "method" => "GET",
      "params" => {},
    })
    assert_nil err
    assert result["ok"]
    assert_equal 200, Helpers.to_int(result["status"])
    assert !result["data"].nil?

    unless setup[:live]
      if result["data"].is_a?(Hash)
        assert_equal "direct01", result["data"]["id"]
      end
      assert_equal 1, setup[:calls].length
    end
  end

end


def v2018_direct_setup(mockres)
  Runner.load_env_local

  calls = []

  env = Runner.env_override({
    "AAREGURU_TEST_V_____ENTID" => {},
    "AAREGURU_TEST_LIVE" => "FALSE",
    "AAREGURU_APIKEY" => "NONE",
  })

  live = env["AAREGURU_TEST_LIVE"] == "TRUE"

  if live
    merged_opts = {
      "apikey" => env["AAREGURU_APIKEY"],
    }
    client = AareguruSDK.new(merged_opts)
    return {
      client: client,
      calls: calls,
      live: true,
      idmap: {},
    }
  end

  mock_fetch = ->(url, init) {
    calls.push({ "url" => url, "init" => init })
    return {
      "status" => 200,
      "statusText" => "OK",
      "headers" => {},
      "json" => ->() {
        if !mockres.nil?
          return mockres
        end
        return { "id" => "direct01" }
      },
      "body" => "mock",
    }, nil
  }

  client = AareguruSDK.new({
    "base" => "http://localhost:8080",
    "system" => {
      "fetch" => mock_fetch,
    },
  })

  {
    client: client,
    calls: calls,
    live: false,
    idmap: {},
  }
end
